sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_train.py

```diff
@@ -73,6 +74,60 @@ def test_train_evo2_runs(tmp_path, num_steps=5):
     assert result.returncode == 0, "train_evo2 command failed."
 
 
+@pytest.mark.timeout(256)  # Optional: fail if the test takes too long.
+@pytest.mark.slow
+def test_train_evo2_stops(tmp_path, num_steps=500000, early_stop_steps=3):
+    """
+    This test runs the `train_evo2` command with mock data in a temporary directory.
```

Comment:
if you would use train test definition from  test_train_single_gpu it is a bit quicker since subprocess import the whole codebase and it takes a few sec